pylint/checkers/refactoring/refactoring_checker.py

```diff
@@ -1998,16 +1998,18 @@ def _has_return_in_siblings(node: nodes.NodeNG) -> bool:
             next_sibling = next_sibling.next_sibling()
         return False
 
-    def _is_function_def_never_returning(self, node: nodes.FunctionDef) -> bool:
+    def _is_function_def_never_returning(
+        self, node: nodes.FunctionDef | astroid.BoundMethod
+    ) -> bool:
         """Return True if the function never returns, False otherwise.
 
         Args:
-            node (nodes.FunctionDef): function definition node to be analyzed.
+            node (nodes.FunctionDef or astroid.BoundMethod): function definition node to be analyzed.
 
         Returns:
             bool: True if the function never returns, False otherwise.
         """
-        if isinstance(node, nodes.FunctionDef) and node.returns:
+        if isinstance(node, (nodes.FunctionDef, astroid.BoundMethod)) and node.returns:
```

Comment:
```suggestion
        if isinstance(node, (nodes.FunctionDef, nodes.BoundMethod)) and node.returns:
```

Reply:
Applying this suggestion raises `AttributeError: module 'astroid.nodes' has no attribute 'BoundMethod'`, with: 

```console
$ pylint --version
pylint 3.0.0b1
astroid 3.0.0a4
Python 3.8.16 (default, Jun  7 2023, 02:00:15) 
[GCC 11.3.0]
```

Reply:
My bad, not sure if we import with ``bases.B...`` elsewhere in the code though. nvm

Reply:
We do, but for some reason the `astroid...` form is more prevalent. IDK.